src/d3a/models/powerflow/__init__.py

```diff
@@ -0,0 +1,132 @@
+            area.ess_device = self._add_storage_device(area)
+        elif isinstance(area.strategy, PVStrategy):
+            area.pv_device = \
+                self._add_generation_device(area, area.strategy.max_panel_power_W * 1e-3)
+        elif isinstance(area.strategy, FinitePowerPlant):
+            area.power_plant = \
+                self._add_generation_device(area, area.strategy.max_available_power_kW)
+        for child in area.children:
+            self._grid_forming(child)
+
+    def _create_bus(self, area, voltage):
+        """
+        :param area: area node where bus bar needs to be added
+        :param voltage: voltage level of bus bar
+        :return: It will add a bus-bar to the specified area node
+        """
+        return pp.create_bus(self.network, vn_kv=voltage, name=area.name)
+
+    def _add_external_grid(self, area):
+        """
+        :param area: area node where external grid element needs to be added
+        :return: It will return an external grid device like MarketMaker
+        """
+        return pp.create_ext_grid(self.network, bus=area.parent.bus, vm_pu=1.0, name=area.name)
+
+    def _add_load_device(self, area, avg_power_W):
+        """
+        :param area: Takes area that is using Load as a strategy
+        :param avg_power_W: Peak power of the load device
+        :return: It will add a load device to the specified area node
+        """
+        return pp.create_load(self.network, bus=area.parent.bus, p_mw=avg_power_W * 1e-6,
+                              name=area.name)
+
+    def _add_generation_device(self, area, peak_power_kw):
+        """
+
+        :param area: Takes area that is using generating strategy
+        :param peak_power_kw: Takes into account power rating of the generating device
+        :return: It will add a generating device to the specified area node
+        """
+        return pp.create_gen(self.network, bus=area.parent.bus, p_mw=peak_power_kw * 1e-3,
+                             name=area.name)
+
+    def _add_storage_device(self, area):
+        """
+        :param area: Takes area that is using Storage as a strategy
+        :return: It will add an Storage device to the specified area node
+        """
+        soc_ratio = area.strategy.initial_soc/100
+        battery_capacity_mwh = area.strategy.state.capacity * 1e-3
+        min_energy_mwh = area.strategy.state.min_allowed_soc_ratio * battery_capacity_mwh
+        return pp.create_storage(self.network, bus=area.parent.bus, p_mw=battery_capacity_mwh,
+                                 max_e_mwh=battery_capacity_mwh, min_e_mwh=min_energy_mwh,
+                                 soc_percent=soc_ratio, name=area.name)
+
+    def _add_line(self, source_area, target_area):
+        """
+        :param source_area: parent area of referenced area
+        :param target_area: child area of referenced area
+        :return: It will add an electrical line connecting parent and child area
```

Comment:
Minor, but the "return" member of the doc string is for explaining what is returned and not what the function does. Please move the "It will ..." sentences in this (and other) functions to the top of the doc string, Thanks!